core/src/mledb/mledb-match/mledb-match.service.ts

```diff
@@ -156,4 +168,171 @@ export class MledbMatchService {
+                return "Failure";
+            }
+        
+        } else if (seriesType === SeriesType.Scrim) {
+            if (!series.scrim) {
+                this.logger.error(`Error: series with id=\`${seriesId}\` has no associated scrim. If this is a league play series, please use \`!ncpSeries\` or \`!unNcpSeries\`. If this actually is a scrim, ping Bot Team.`);
+                return "Failure";
+            }
+
+            winningTeam = await this.teamRepo.findOneOrFail({
+                where: {
+                    name: "FA",
+                },
+            });
+        } else {
+            throw new Error();
+        }
+
+        const seriesReplays = await this.seriesReplayRepo.find({
+            where: {
+                series: {
+                    id: seriesId,
+                },
+            },
+        });
+
+        // Set series to Full NCP and set winning team
+        series.fullNcp = isNcp;
+        await this.seriesRepo.save(series);
+
+        // Update each series replay (including any dummies) to NCP
+        const replayIds = seriesReplays.map(seriesReplay => seriesReplay.id);
+        await this.markReplaysNcp(replayIds, isNcp, winningTeam ? winningTeam : undefined);
+
+        this.logger.debug(`(${r}) end markSeriesNcp`);
+
+        return `seriesId=${seriesId} successfully marked fullNcp=${isNcp} with updated elo, and all connected replays had their elo updated.`;
+    }
+
+    /**
+     * Marks replays as NCP=true/false, and updates the associated Elo of those replays and all connected replays accordingly.
+     * "Connected" replays are where replays in which one of the player's in the NCP replay has played. Since the NCP replay will have its elo affects removed,
+     * all subsequent replays where those player's played need to be recalculated.
+     * @param replayId The replay to mark NCP=true/false.
+     * @param isNcp Whether the given replayId should be marked NCP or un-NCP.
+     * @returns A string containing status of what was updated.
+     */
+    async markReplaysNcp(replayIds: number[], isNcp: boolean, winningTeam?: MLE_Team): Promise<string> {
+        const r = Math.floor(Math.random() * 10000);
+        this.logger.debug(`(${r}) begin markReplaysNcp: replayIds=${replayIds}, isNcp=${isNcp}, winningTeam=${winningTeam}`);
+
+        // Winning team must be specified if NCPing replays
+        if (isNcp && !winningTeam) return "Can't mark an NCP without a winning team";
+
+        // Make sure we are considering replayIds in chronological order
+        replayIds.sort((a, b) => a - b);
+
+        // Gather replays
+        const replayPromises = replayIds.map(async rId => this.seriesReplayRepo.findOneOrFail({
+            where: {
+                id: rId,
+            },
+            relations: ["teamCoreStats.teamName"],
```

Comment:
Use objects for relations, not an array of strings. 
```suggestion
            relations: {
                        teamCoreStats: true,
            },
```